src/features/klant/bedrijf/BedrijfZoeker.vue

```diff
@@ -109,18 +138,37 @@ const currentQuery = computed(() => {
   });
 });
 
+const currentKlantQuery = computed(() => {
+  const { currentSearch, field } = state.value;
+
+  if (field === "telefoonnummer" || field === "email")
+    return createKlantQuery({
+      field,
+      query: currentSearch,
+    });
+
+  return undefined;
+});
+
 watch(
-  [currentQuery, inputRef],
+  [currentBedrijfQuery, inputRef],
   ([query, input]) => {
     if (!(input instanceof HTMLInputElement)) return;
     input.setCustomValidity(query instanceof Error ? query.message : "");
   },
   { immediate: true }
 );
 
+const klanten = useSearchKlanten({
+  query: computed(() => state.value.klantSearchQuery),
```

Comment:
best abstract en indirect: handlesearch verandert een state value, waardoor een computed prop iets in useSearchklanten iets triggert. is die complexiteit nodig? Waarschijnlijk te simpel gedacht, maar waarom niet gewoon zoiets: 
`const handleSearch = () => klantenservice.search( bedrijfsnaam, ... )`

Reply:
Ik weet niet meer precies waarom we voor deze oplossing zijn gegaan, maar ik vermoed dat het te maken heeft met enerzijds de stale-while-revalidate cache, en anderzijds het bewaren van de zoekopdracht (voor als je heen en weer navigeert). Dit is hoe het al werkte, ik heb het alleen meer in lijn gebracht met de PersoonZoeker. Dat was nodig omdat we nu in twee verschillende bronnen zoeken (de gateway was ingericht alsof je in de KvK kan zoeken op telefoonnummer/email-adres)

Reply:
als de aanpassing is om het meer in lijn te brengen met persoonzoeker dan is het goed